x-pack/plugins/ml/server/lib/alerts/alerting_service.ts

```diff
@@ -397,6 +398,25 @@ export function alertingServiceProvider(
     return alertInstanceKey;
   };
 
+  const getAlertMessage = (
+    resultType: MlAnomalyResultType,
+    source: Record<string, unknown>
+  ): string => {
+    let message = i18n.translate('xpack.ml.alertTypes.anomalyDetectionAlertingRule.alertMessage', {
+      defaultMessage:
+        'Alerts are raised based on real-time scores. Remember that scores may be adjusted over time as data continues to be analyzed.',
+    });
+
+    if (resultType === ML_ANOMALY_RESULT_TYPE.RECORD) {
+      message = getMetricChangeDescription(
+        source.actual as number[],
```

Comment:
For population jobs with a `causes` field, the `actual` and `typical` values will be nested inside the `causes` field. You will need to do a similar check here as we use in the anomalies table, and copy the `actual` and `typical` to top level fields if these is a single `cause`. For records where there is more than 1 cause you should use the default message.

Even when it is set, the message will just indicate how the `actual` compares to the `typical` e.g. `6x higher`. I think this needs to be incorporated into a longer string of text, for example

```
Anomaly in High total sales found for customer_full_name.keyword Sultan Al Bryan where actual is 6x higher than typical.
```

But some extra logic will be needed to handle all the various formats that `message` can take

```
2x higher
5x lower
More than 100x higher
More than 100x lower
Unusually low
Unusually high
Unexpected zero value
Unexpected non-zero value
```

Reply:
Updated in https://github.com/elastic/kibana/pull/172473/commits/1a487f26ee9e9430a7ea537ffe6a93c54938b7d6